src/jobs/integration-test.yaml

```
@@ -0,0 +1,43 @@
+description: |
+  Runs an integration test by creating a KIND cluster and executing it as a Go
+  test.
+
+  Uses the machine executor and replaces the Go version with a version
+  controlled by us since the preinstalled Go is very outdated.
+
+  Tests can be customized by providing a KIND config file or a setup script
+  which is run before the test is executed.
+environment:
+  KUBERNETES_VERSION: v1.16.3
+executor: machine
+parameters:
+  env-file:
+    default: ""
+    description: "File of environment variables to set."
+    type: string
+  kind-config:
+    default: ""
+    description: "Path to kind config file."
+    type: string
+  setup-script:
+    default: ""
+    description: "Path to setup script with commands to execute before each test."
+    type: string
+  test-dir:
+    description: "Tested package directory."
+    type: string
+steps:
+  - checkout
+  - attach_workspace:
+      at: .
+  - go-cache-restore
+  - machine-install-go
+  - integration-test-install-tools
+  - integration-test-create-cluster:
+      kind-config: << parameters.kind-config >>
+  - integration-test-setup:
+      setup-script: << parameters.setup-script >>
+  - integration-test-go-test:
+      env-file: << parameters.env-file >>
+      test-dir: << parameters.test-dir >>
+  - integration-test-export-logs
```

How about adding `when: always` here instead of the command itself? I think it makes things clearer from the high level veiw.

```suggestion
  - integration-test-export-logs
    when: always
```

---

Yes makes sense. Done.

---

Nah linter says no. I'll get it sorted.